### DISCUSSION TITLE
Pinning Dagster version in Dagster Cloud Deployment

### DISCUSSION THREAD

**Discussion by JoeNaso** [dagster-io/dagster#17975] [Q&A]
Is it possible to pin the version of Dagster being used in a Dagster Cloud Deployment? 

My setup.py file specifies an approximate version, but looking at the downloaded Debug file, I see a different version being used in Dagster Cloud. I am trying to narrow down the possibility that an API change between those releases is causing issues with MultiPartitioned Assets deployed to Prod that Im not seeing in development on local

```
# debug file
...    
  },
    "version": "1.5.7"
}
```
```
# setup.py
...
    install_requires=[
        "dagster~=1.3.0",
...
```

**Comment by prha** [comment]
You can pin your individual code locations to different dagster versions in Dagster Cloud.  Dagster Cloud supports backwards compatibility with dagster versions >= `0.13.2`.  The hosted web UI will always pinned to the most recent version of dagster, so the version in any debug file will always reflect that.

**Reply by JoeNaso** [reply]
Does that mean there are potentially 3 versions involved in a single project? 

The pinned dependency, the version running on Dagster Cloud (presumably a serverless deployment), and the UI version? And the UI has no relation to the pinned project version?

**Reply by prha** [reply]
The UI version and the serverless agent will always be running the most recent version of Dagster.  The individual code locations which host your definitions can all have different pinned project versions.  If you only have one code location for your project, you typically are only dealing with 2 versions: one for the definitions / runs, and one for the cloud infrastructure (UI, daemons, agent).

This [architecture diagram](https://docs.dagster.io/dagster-cloud/deployment/hybrid#hybrid-architecture-overview) is displaying a breakdown of components for Hybrid, but describes Serverless setup in that all of the components labeled "User code" are running the pinned dagster version and all the other components (the Agent, and the entire grouping labeled "Dagster Cloud") are running the most recent version.
